File: frontend/app/models/folder.js
import Model, { attr, hasMany, belongsTo } from "@ember-data/model";

export default class Folder extends Model {
  @attr("string") name;
  @attr("string") description;
  @attr("number") unreadTransferredCount;
  @hasMany("encryptable", { async: false, inverse: "folder" }) encryptables;
  @belongsTo("team", { async: false, inverse: "folders" }) team;

  get isInboxFolder() {
    return this.name === "inbox" && this.team.get("isPersonalTeam");
  }
}

File: app/models/folder.rb
# frozen_string_literal: true

# == Schema Information
#
# Table name: folders
#
#  id          :integer          not null, primary key
#  name        :string(40)       default(""), not null
#  description :text
#  created_at  :datetime         not null
#  updated_at  :datetime         not null
#  team_id     :integer          default(0), not null
#  personal_inbox :boolean       default(false)
#

class Folder < ApplicationRecord
  belongs_to :team
  has_many :encryptables, dependent: :destroy

  attr_readonly :team_id

  validates :name, presence: true
  validates :name, uniqueness: { scope: :team }
  validates :name, length: { maximum: 70 }
  validates :description, length: { maximum: 300 }
  validates :personal_inbox, uniqueness: { scope: :team, if: :personal_inbox? }

  def label
    name
  end

  def unread_count_transferred_encryptables
    encryptables.where.not(encrypted_transfer_password: nil).count
  end
end
